packages/salesforcedx-vscode-core/src/commands/forceSourceDeploy.ts

```diff
@@ -5,37 +5,129 @@
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-import {
-  Command,
-  SfdxCommandBuilder
-} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { CliCommandExecutor, Command, SfdxCommandBuilder } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as path from 'path';
+import { Observable } from 'rxjs/Observable';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
 import { nls } from '../messages';
-import {
-  SfdxCommandlet,
-  SfdxCommandletExecutor,
-  SfdxWorkspaceChecker
-} from './commands';
-import {
-  FileType,
-  ManifestOrSourcePathGatherer,
-  SelectedPath
-} from './forceSourceRetrieve';
-
-export class ForceSourceDeployExecutor extends SfdxCommandletExecutor<
-  SelectedPath
-> {
+import { notificationService, ProgressNotification } from '../notifications';
+import { taskViewService } from '../statuses';
+import { SfdxCommandlet, SfdxCommandletExecutor, SfdxWorkspaceChecker } from './commands';
+import { FileType, ManifestOrSourcePathGatherer, SelectedPath } from './forceSourceRetrieve';
+
+interface Result {
```

Comment:
Also, these types are only for apex and I'm not sure they are even complete in that regard...

Reply:
I think these interfaces and the parsing methods should be moved to the `salesforcedx-utils-vscode` module under `cli` since we can potentially use them for parsing results from other command executions.

As mentioned below, I've tested errors on different metadata error results but will get back to you on what changes might be needed for the interfaces.